src/app/qgswelcomepage.cpp

```diff
@@ -187,10 +209,45 @@ void QgsWelcomePage::showContextMenuForProjects( QPoint point )
   QAction *removeProjectAction = new QAction( tr( "Remove from List" ), menu );
   connect( removeProjectAction, &QAction::triggered, this, [this, index]
   {
-    mModel->removeProject( index );
+    mRecentProjectsModel->removeProject( index );
     emit projectRemoved( index.row() );
   } );
   menu->addAction( removeProjectAction );
 
   menu->popup( mapToGlobal( point ) );
+  connect( menu, &QMenu::aboutToHide, menu, &QMenu::deleteLater );
+}
+
+void QgsWelcomePage::showContextMenuForTemplates( QPoint point )
+{
+  QModelIndex index = mTemplateProjectsListView->indexAt( point );
+  if ( !index.isValid() )
+    return;
+
+  QFileInfo fileInfo( index.data( QgsRecentProjectItemsModel::NativePathRole ).toString() );
+
+  QMenu *menu = new QMenu();
```

Comment:
Leaks?

Reply:
Shouldn't https://github.com/qgis/QGIS/blob/9ff115685478a14503b4af60334c859c72e9f6bb/src/app/qgswelcomepage.cpp#L218
I guess it's better to delete it when it's hidden then keeping it around for the whole lifetime of the app